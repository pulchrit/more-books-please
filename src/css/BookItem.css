.book-card {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: space-around;
    padding: 7px;
    margin: 10px;
    border: 1px solid  rgba(10, 34, 73, 0.8);
    border-radius: 3px;
}

.book-cover {
    align-self: center;
    max-width: 150px;
    margin-bottom: 10px;
}

.book-details {
    margin: 4px;
    line-height: 1.2;
}

a.find-link {
    font-size: 1em;
    text-decoration: underline;
    margin: 12px 4px 0 4px;
    cursor: pointer;
}