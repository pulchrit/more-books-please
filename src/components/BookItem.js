import React from 'react';
import '../css/BookItem.css';

const BookItem = ({rank, title, author, description, image, url}) => {
    return (
        <section className="book-card">
            <img className="book-cover" src={image} alt={`${title} by ${author}`} />
            <p className='book-details'>{`Number: ${rank}`}</p>
            <p className='book-details'>{title}</p>
            <p className='book-details'>{`By ${author}`}</p>
            <p className='book-details'>{description}</p>
            <a className='find-link' href={url}>Find this book</a>
        </section>
    );
}

export default BookItem;