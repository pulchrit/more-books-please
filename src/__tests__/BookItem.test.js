import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import BookItem from '../components/BookItem';

describe ("BookItem", () => {

    const book = 
        {
            rank: '1',
            primary_isbn10: "0385376715",
            description: "A celebration of future possibilities.",
            title: "The Wonderful Things You Will Be",
            author: "Emily Winfield Martin",
            book_image: "https://s1.nyt.com/du/books/images/9780385376716.jpg",
            open_library_url: "https://openlibrary.org/books/OL26629977M/The_Wonderful_Things_You_Will_Be",
            amazon_product_url: "http://www.amazon.com/The-Wonderful-Things-You-Will/dp/0385376715?tag=NYTBS-20"
        };


  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <BookItem 
            key={book.rank}
            rank={book.rank}
            title={book.title}
            author={book.author}
            description={book.description}
            image={book.book_image}
            url={book.open_library_url === '' ? book.amazon_product_url : book.open_library_url}
        /> 
      </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
            <BookItem 
                key={book.rank}
                rank={book.rank}
                title={book.title}
                author={book.author}
                description={book.description}
                image={book.book_image}
                url={book.open_library_url === '' ? book.amazon_product_url : book.open_library_url}
            />
        </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })

})


