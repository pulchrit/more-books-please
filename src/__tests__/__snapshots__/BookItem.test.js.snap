// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookItem renders UI as expected 1`] = `
<section
  className="book-card"
>
  <img
    alt="The Wonderful Things You Will Be by Emily Winfield Martin"
    className="book-cover"
    src="https://s1.nyt.com/du/books/images/9780385376716.jpg"
  />
  <p
    className="book-details"
  >
    Number: 1
  </p>
  <p
    className="book-details"
  >
    The Wonderful Things You Will Be
  </p>
  <p
    className="book-details"
  >
    By Emily Winfield Martin
  </p>
  <p
    className="book-details"
  >
    A celebration of future possibilities.
  </p>
  <a
    className="find-link"
    href="https://openlibrary.org/books/OL26629977M/The_Wonderful_Things_You_Will_Be"
  >
    Find this book
  </a>
</section>
`;
